syntax = "proto3";

package swim;

service FailureDetector {
  rpc Ping (PingRequest) returns (PingAck);
  rpc IndirectPing (IndirectPingRequest) returns (IndirectPingAck);
}

message PingRequest {
  string sender_id = 1;
}

message PingAck {
  bool ack = 1;
}

message IndirectPingAck {
    string target_id = 1;
    bool ack = 2;
  }
  
message IndirectPingRequest {
  string sender_id = 1;
  string target_id = 2;
}